File: src/main/java/com/googongill/aditory/controller/dto/user/SignupResponse.java
package com.googongill.aditory.controller.dto.user;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.googongill.aditory.service.dto.category.CategoryResult;
import com.googongill.aditory.service.dto.user.SignupResult;
import lombok.Builder;

import java.util.ArrayList;
import java.util.List;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class SignupResponse {
    private Long userId;
    private String nickname;
    @Builder.Default
    private List<CategoryResult> userCategories = new ArrayList<>();

    public static SignupResponse of(SignupResult signupResult) {
        return SignupResponse.builder()
                .userId(signupResult.getUserId())
                .nickname(signupResult.getNickname())
                .userCategories(signupResult.getUserCategories())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/user/response/SignupResponse.kt
package com.googongill.aditory.controller.dto.user.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.googongill.aditory.service.dto.category.CategoryResult
import com.googongill.aditory.service.dto.user.SignupResult

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class SignupResponse(
    val userId: Long?,
    val nickname: String?,
    val userCategories: List<CategoryResult> = emptyList()
) {

    companion object {
        fun of(signupResult: SignupResult): SignupResponse {
            return SignupResponse(
                userId = signupResult.userId,
                nickname = signupResult.nickname,
                userCategories = signupResult.userCategories
            )
        }
    }
}
